### PR TITLE
Using GzipCompressorInputStream for decompressing .gz files.

### PR DESCRIPTION
**Pull request by sarveshkaushal**
Very small change. This is for another TODO comment.

### CODE REVIEW

File: util/src/main/java/io/kubernetes/client/Copy.java [hunk 1]
```diff
@@ -121,8 +122,7 @@ public void copyDirectoryFromPod(
             false,
             false);
     try (InputStream is = new Base64InputStream(new BufferedInputStream(proc.getInputStream()));
-        ArchiveInputStream archive = new TarArchiveInputStream(is)) {
-      // TODO Use new GzipCompressorInputStream(is))) here {
+        ArchiveInputStream archive = new TarArchiveInputStream(new GzipCompressorInputStream(is))) {
```
Comment (yue9944882) [inline util/src/main/java/io/kubernetes/client/Copy.java]: i think we need to enable the gzip compression on the remote process too, by applying "z" option? additionally, we need a test for the change.

Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/Copy.java]: Yep, we definitely need it on both sides, the current encoding is just 'tar'

Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/Copy.java]: I think
final Process proc =
        this.exec(
            namespace,
            pod,
            new String[] {"sh", "-c", "tar cf**z** - " + srcPath + " | base64"},
            container,
            false,
            false);
Am I thinking right?

**Comment by sarveshkaushal** [comment]
I have added a test case for the change. Also enabled zipping on remote process. Please take a look.

File: util/src/main/java/io/kubernetes/client/Copy.java (hunk 1 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/Copy.java]: yes, but you dropped "f", is that intended?

File: util/src/test/java/io/kubernetes/client/CopyTest.java [hunk 2]
```diff
@@ -127,4 +129,53 @@ public void run() {
...
+
+    Copy copy = new Copy(client);
+
+    stubFor(
+        get(urlPathEqualTo("/api/v1/namespaces/" + namespace + "/pods/" + podName + "/exec"))
+            .willReturn(
+                aResponse()
+                    .withStatus(404)
+                    .withHeader("Content-Type", "application/json")
+                    .withBody("{}")));
```
Comment (yue9944882) [inline util/src/test/java/io/kubernetes/client/CopyTest.java]: ahh, so sry for oversight.. the exec call works basing on websocket, there's no way to mock that out via wiremock framework.. i will verify this pull locally. we should remove the test

File: util/src/main/java/io/kubernetes/client/Copy.java (hunk 1 above)
Comment (sarveshkaushal) [reply util/src/main/java/io/kubernetes/client/Copy.java]: Yes- we don't need "f". As we are directly getting the stream from process' stdout.

File: util/src/test/java/io/kubernetes/client/CopyTest.java (hunk 2 above)
Comment (sarveshkaushal) [reply util/src/test/java/io/kubernetes/client/CopyTest.java]: Let me remove the test.

Comment (sarveshkaushal) [reply util/src/test/java/io/kubernetes/client/CopyTest.java]: Test case is removed.

**Comment by brendandburns** [comment]
This looks fine to me. Is it covered by existing test code?

**Comment by sarveshkaushal** [comment]
I see a test case for method copyFileToPod but don't see any other test case for copyDirectoryFromPod. However currently test for copyFileToPod is ignored because it was causing instability in the build.

File: util/src/main/java/io/kubernetes/client/Copy.java (hunk 1 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/Copy.java]: iiuc, if you drop "f", you will also have to drop the tailing "-" from the command. in the previous command `tar cf - ...` where "-" is the argument for "f" from the bundled options. i suggest keep the "f" for clarification given that it's a more commonly used form.

**Comment by brendandburns** [comment]
ok, we can merge this as is. We should fix the tests too...

**Comment by sarveshkaushal** [comment]
Thanks @brendandburns .I can take a look at test case for copyFileToPod but need to little more details about build. I will investigate and create a separate PR for it.

**Comment by yue9944882** [comment]
> new String[] {"sh", "-c", "tar cfz - " + srcPath + " | base64"},

https://github.com/kubernetes-client/java/pull/754#discussion_r339897994 i dont think this is a legitimate tar command

**Pull request merged**